menuconfig.py

```diff
@@ -267,123 +267,104 @@ def style_from_string(sstr):
         "default":  -1
     }
 
-    def __parse_tokenizer(cdef):
-        for cs in cdef.replace(' ', '_').split('_'):
-            if cs == '': continue
-            yield cs
+    def __parse_color(t):
+        ts = t.split(":")
+        if len(ts) != 2:
+            raise StyleError("Invalid color notation: {}".format(t))
 
-        yield None
+        cdef = ts[1]
 
-    def __parse_color(tok, t):
-        bright = False
+        if cdef in _STYLE_STD_COLORS:
+            return _STYLE_STD_COLORS[cdef]
 
-        if t == 'bright':
-            bright = True
-            t = next(tok)
-
-        if t in __STYLE_STD_COLORS:
-            color = __STYLE_STD_COLORS[t]
-            if bright:
-                color += 8
-            return color
-
-        elif t.startswith('color'):
-            if bright:
-                raise StyleError("Fixed color cannot have a bright attribute. Error near: {}".format(t))
-
-            cnum = t[len('color'):]
+        elif cdef.startswith('color'):
+            cnum = cdef[len('color'):]
             if not cnum.isdigit():
                 raise StyleError('Fixed color must be followed by a number. Error near: {}'.format(t))
 
             cnum = int(cnum)
             if cnum >= curses.COLORS or cnum < -1:
-                raise StyleError("Fixed color must be in range -1..{}. Error near: {}".format(curses.COLORS, t))
+                raise StyleError("Fixed color must be in range 0..{}. Error near: {}".format(curses.COLORS - 1, t))
 
             return int(cnum)
 
         else:
-            raise StyleError("Unknown token: {}".format(t))
-
-    def __parse_main(tok):
-        nonlocal attrs, fg_color, bg_color
-
-        # Parse attributes
-        while True:
-            t = next(tok)
-
-            if t == None:
-                return
-            if t == 'bold':
-                attrs |= curses.A_NORMAL if _IS_WINDOWS else curses.A_BOLD
-            elif t == 'standout':
-                attrs |= curses.A_STANDOUT
-            elif t == 'underline':
-                attrs |= curses.A_UNDERLINE
-            else:
-                break
-
-        # Parse the color
-        fg_color = __parse_color(tok, t)
-
-        t = next(tok)
-
-        # No more tokens, exit
-        if not t:
-            return
-        # Do we have a background color?
-        elif t != "on":
-            raise StyleError("A foreground color must be followed by \"on\" to define a background color. Error near: {}".format(t))
-
-        t = next(tok)
-        bg_color = __parse_color(tok, t)
-
-        t = next(tok)
-        if t != None:
-            raise StyleError("Garbage at end of string. Error near: {}".format(t))
+            raise StyleError("Unknown color definition: {}".format(t))
 
     attrs = 0
     fg_color = -1
     bg_color = -1
 
-    __parse_main(__parse_tokenizer(cstr))
+    # Parse attributes
+    for t in cstr.split(','):
+        if t == '': continue
+
+        if t == 'bold':
```

Comment:
My preferred quoting style is `'bold'`.

Or to be honest, I don't really have a strong preference, but it's what the rest of the code already uses, and consistency is nice. :)